staging/src/k8s.io/apiserver/pkg/util/webhook/authentication.go

```diff
@@ -31,6 +33,37 @@ import (
 // rest.Config generated by the resolver.
 type AuthenticationInfoResolverWrapper func(AuthenticationInfoResolver) AuthenticationInfoResolver
 
+// NewDefaultAuthenticationInfoResolverWrapper builds a default authn resolver wrapper
+func NewDefaultAuthenticationInfoResolverWrapper(
+	proxyTransport *http.Transport,
+	loopbackClientConfig *rest.Config) AuthenticationInfoResolverWrapper {
+
+	webhookAuthResolverWrapper := func(delegate AuthenticationInfoResolver) AuthenticationInfoResolver {
+		return &AuthenticationInfoResolverDelegator{
+			ClientConfigForFunc: func(server string) (*rest.Config, error) {
+				if server == "kubernetes.default.svc" {
+					return loopbackClientConfig, nil
+				}
+				return delegate.ClientConfigFor(server)
+			},
+			ClientConfigForServiceFunc: func(serviceName, serviceNamespace string) (*rest.Config, error) {
+				if serviceName == "kubernetes" && serviceNamespace == corev1.NamespaceDefault {
```

Comment:
the hard-coding looks odd tho 😥but it should be another issue

Reply:
> the hard-coding looks odd tho but it should be another issue

The hardcoding is based on the idea that the kube apiserver (in that hardcoded location in a cluster), can be trusted with a token intended for the API.  Since SA tokens are auto-mounted and intended for the API, that host is safe to trust.

Reply:
then we shouldnt move these hardcoding (L44 also) into generic apiserver, amright? since it only applies to kube apiserver.